### PR TITLE
Group Warning Icons + Warning Bug Fixes

### PR DESCRIPTION
**Pull request by OliverEGreen**
### Purpose

This PR adds warning/error icons to groups when they contain nodes (or nested groups with nodes) in a warning/error state. 
Note: Groups will still display these icons even if the only warnings they contain are dismissed. This is intentionally done so that warnings cannot be hidden inside of groups, making them hard to find for users. 

- This PR also updates the MissingNode.png to a larger image resolution so as to appear smoother.
- This PR also fixes a bug with warnings not appearing for certain nodes (e.g. missing nodes). 

FxeGY7lyP9

### Declarations

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Adds warning/error icons to group headers.

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 1]
```diff
@@ -329,6 +332,20 @@ public GeometryCollection NestedGroupsGeometryCollection
...
         }
 
+        /// <summary>
+        /// Whether the group contains nodes that are in an info/warning/error state.
+        /// This includes any nodes that are in nested groups.
+        /// </summary>
+        public ElementState GroupState
+        {
+            get => groupState;
+            set
```
Comment (SHKnudsen) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs]: `internal` if possible

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 2]
```diff
@@ -949,6 +968,9 @@ private void model_PropertyChanged(object sender, System.ComponentModel.Property
                 case "FontSize":
                     RaisePropertyChanged("FontSize");
                     break;
+                case nameof(NodeModel.State):
+                    UpdateErrorAndWarningIconVisibility();
+                    break;
```
Comment (SHKnudsen) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs]: Should this be removed?

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/DummyNode.cs [hunk 3]
```diff
@@ -14,13 +15,18 @@ public void CustomizeView(DummyNode model, Dynamo.Controls.NodeView nodeView)
             if (model.NodeNature == DummyNode.Nature.Unresolved)
                 fileName = "MissingNode.png";
 
+            model.State = ElementState.Warning;
```
Comment (SHKnudsen) [inline src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/DummyNode.cs:18]: why are we modifying this?

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 4]
```diff
@@ -2,6 +2,7 @@
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Collections.Specialized;
+using System.ComponentModel;
```
Comment (SHKnudsen) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:5]: what is this used for?

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/DummyNode.cs (hunk 3 above)
Comment (OliverEGreen) [reply src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/DummyNode.cs:18]: The DummyNode is a bit weird, it doesn't seem to want to recognise it's in a warning state for some reason. 
It only appears due to an unresolved node, so thought I'd just force it.

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs (hunk 4 above)
Comment (OliverEGreen) [reply src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs:5]: We're using PropertyChangedEventArgs when subscribing to the NodeModel.State change

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs (hunk 2 above)
Comment (OliverEGreen) [reply src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs]: True - removed!

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs (hunk 1 above)
Comment (OliverEGreen) [reply src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs]: yep done

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 5]
```diff
@@ -329,6 +332,20 @@ public GeometryCollection NestedGroupsGeometryCollection
             get => new GeometryCollection(GroupIdToCutGeometry.Values.Select(x => x));
         }
 
+        /// <summary>
+        /// Whether the group contains nodes that are in an info/warning/error state.
```
Comment (radumg) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs]:
```suggestion
        /// Indicates whether the group contains nodes that are in an info/warning/error state.
```

File: src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs [hunk 6]
```diff
@@ -329,6 +332,20 @@ public GeometryCollection NestedGroupsGeometryCollection
...
+        /// This includes any nodes that are in nested groups.
```
Comment (radumg) [inline src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs]:
```suggestion
        /// This includes the state of any nodes that are in nested groups.
```

**Pull request closed without merging**